aptos-move/framework/aptos-framework/sources/voting.move

```diff
@@ -299,6 +343,9 @@ module aptos_framework::voting {
         let proposal = table::borrow_mut(&mut voting_forum.proposals, proposal_id);
         assert!(!proposal.is_resolved, error::invalid_state(EPROPOSAL_ALREADY_RESOLVED));
 
+        let multi_step_key = utf8(IS_MULTI_STEP_PROPOSAL_KEY);
+        assert!(!simple_map::contains_key(&proposal.metadata, &multi_step_key) || *simple_map::borrow(&proposal.metadata, &multi_step_key) == to_bytes(&false), error::permission_denied(EMULTI_STEP_PROPOSAL_CANNOT_USE_SINGLE_STEP_RESOLVE_FUNCTION));
```

Comment:
nit: This is a bit hard to read. We can declare two variables here to improve readability:

// Default to treating the proposal as single-step if the is_multi_step_proposal attribute is not present.
let has_multi_step_attribute = !simple_map::contains_key(&proposal.metadata, &multi_step_key);
let is_multi_step_proposal = from_bcs::to_bool(*simple_map::borrow(&proposal.metadata, &multi_step_key));
assert!(!has_multi_step_attribute || !is_multi_step_proposal, ...);